xt_training/utils/functional.py

```diff
@@ -89,11 +91,11 @@ def train(
         for loader_name, loader in test_loaders.items():
             runner(loader, loader_name)
 
-    best_loss = 1e12
-    if val_loader:
+    best_loss = 1e12 if starting_loss is None else starting_loss
+    if val_loader and starting_loss is None:
```

Comment:
How about we just set the default input to `best_loss=1e12`, then we can remove line 94 and change line 95 to `if val_loader:`

Reply:
I've set it up this way because I don't want it to calculate the validation loss before training since I already provide the starting loss. I did this because when doing k-fold cross validation it was calculating the initial loss k times, which was slow.

I still want to provide `val_loaders` so that it can calculate the new validation loss after training though, and there wasn't any way to skip the initial validation loss while including `val_loaders`.

Maybe we should control that with a separate flag to be more clear.

Reply:
Ah got it. Good as it is then I reckon